client_handlers.go

```diff
@@ -5,64 +5,82 @@ import (
 	"fmt"
 )
 
-func clientHandshakeHandler(c *Conn) (*alert, error) {
-	initalizeCipherSuite := func(h *handshakeMessageServerKeyExchange) (*alert, error) {
-		clientRandom, err := c.state.localRandom.Marshal()
+func initalizeCipherSuite(c *Conn, h *handshakeMessageServerKeyExchange) (*alert, error) {
+	clientRandom, err := c.state.localRandom.Marshal()
+	if err != nil {
+		return &alert{alertLevelFatal, alertInternalError}, err
+	}
+	serverRandom, err := c.state.remoteRandom.Marshal()
+	if err != nil {
+		return &alert{alertLevelFatal, alertInternalError}, err
+	}
+
+	if c.state.extendedMasterSecret {
+		var sessionHash []byte
+		sessionHash, err = c.handshakeCache.sessionHash(c.state.cipherSuite.hashFunc())
```

Comment:
`sessionHash, err := c.handshakeCache.sessionHash(c.state.cipherSuite.hashFunc())`

Reply:
so you don't need to do `var sessionHash []byte`

Reply:
Fixed!